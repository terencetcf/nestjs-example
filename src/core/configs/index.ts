export * from './app.config';
export * from './http-server.config';
