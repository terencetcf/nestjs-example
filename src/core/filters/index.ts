export * from './exception.filter';
export * from './global.filter';
export * from './http-exception.filter';
