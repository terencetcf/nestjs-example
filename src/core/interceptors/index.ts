export * from './global.interceptor';
export * from './response.interceptor';
