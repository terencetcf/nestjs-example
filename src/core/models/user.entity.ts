import { Column, Entity, PrimaryColumn } from 'typeorm';

@Entity()
export class User {
  @PrimaryColumn({ type: 'uuid' })
  id!: string;

  @Column({
    length: 50,
  })
  username!: string;

  @Column({
    length: 100,
  })
  first_name!: string;

  @Column({
    length: 100,
  })
  last_name!: string;
}
