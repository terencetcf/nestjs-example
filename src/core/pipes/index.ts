export * from './global.pipe';
export * from './parse-number-id.pipe';
