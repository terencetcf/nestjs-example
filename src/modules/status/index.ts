export * from './dto';
export * from './status.controller';
export * from './status.module';
export * from './status.service';
