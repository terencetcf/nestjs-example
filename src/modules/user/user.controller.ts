import { Body, Controller, Get, HttpStatus, Post } from '@nestjs/common';
import { ApiOperation, ApiResponse } from '@nestjs/swagger';

import { CreateUserDto, UserDto } from './dto';
import { UserService } from './user.service';

@Controller()
export class UserController {
  constructor(private readonly userService: UserService) {}

  @ApiOperation({
    summary: 'Get users',
  })
  @ApiResponse({
    status: HttpStatus.OK,
    type: UserDto,
    isArray: true,
    description: 'List of users',
  })
  @Get('users')
  async getUsers(): Promise<UserDto[]> {
    const result = await this.userService.getUsers();

    return result;
  }

  @ApiOperation({
    summary: 'Create user',
  })
  @ApiResponse({
    status: HttpStatus.CREATED,
    description: 'User created',
    type: UserDto,
  })
  @ApiResponse({
    status: HttpStatus.BAD_REQUEST,
    description: `Invalid payload`,
  })
  @Post('users')
  async createUser(@Body() payload: CreateUserDto): Promise<UserDto> {
    const result = await this.userService.createUser(payload);

    return result;
  }
}
