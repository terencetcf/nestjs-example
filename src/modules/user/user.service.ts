import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { User } from '@tt/core/models';
import { generatorUtil } from '@tt/utils';
import { Repository } from 'typeorm';

import { CreateUserDto, UserDto } from './dto';

@Injectable()
export class UserService {
  constructor(
    @InjectRepository(User)
    private readonly userRepository: Repository<User>,
  ) {}

  async getUsers(): Promise<UserDto[]> {
    const users = await this.userRepository.find();

    const dto = users.map((user) => new UserDto(user));

    return dto;
  }

  async createUser(createUserDto: CreateUserDto): Promise<UserDto> {
    const mappedUser: Partial<User> = {
      id: generatorUtil.uuid(),
      username: createUserDto.username,
      first_name: createUserDto.first_name,
      last_name: createUserDto.last_name,
    };

    const createdUser = await this.userRepository.save(mappedUser);

    return new UserDto(createdUser);
  }
}
