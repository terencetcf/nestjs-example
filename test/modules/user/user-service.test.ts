import { User } from '@tt/core/models';
import { UserService } from '@tt/modules/user';
import { TestApp } from '@tt/test/setup';

import { testData } from './__fixtures__';

describe('UserService', () => {
  let app: TestApp;
  let sut: UserService;

  beforeAll(async () => {
    app = await TestApp.create().setup(
      [],
      [UserService],
      [User],
      testData.entitiesToBeAddedToDb,
    );
    sut = app.get(UserService);
  });

  afterAll(async () => {
    await app.close();
  });

  describe('getUsers', () => {
    it('should return expected result', async () => {
      // Act
      const actual = await sut.getUsers();

      // Assert
      expect(actual).toMatchSnapshot();
    });
  });

  describe('createUser', () => {
    it('should return the created company', async () => {
      // Act
      const actual = await sut.createUser(testData.createUserDto);

      // Assert
      expect(actual).toMatchSnapshot({
        id: expect.any(String),
      });
    });
  });
});
