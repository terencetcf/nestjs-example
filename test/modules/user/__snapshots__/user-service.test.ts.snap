// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService createUser should return the created company 1`] = `
Object {
  "first_name": "Steve",
  "id": Any<String>,
  "last_name": "Job",
  "username": "new-username",
}
`;

exports[`UserService getUsers should return expected result 1`] = `
Array [
  UserDto {
    "first_name": "John",
    "id": "070b50ca-486a-41e7-9a2e-cda2c16204ac",
    "last_name": "Doe",
    "username": "test_user",
  },
]
`;
